api/consumer/countDocumentReference/src/v1/handler.py

```diff
@@ -37,43 +30,21 @@ def search_document_references(
     dependencies: FrozenDict[str, Any],
     logger: Logger,
 ) -> PipelineData:
-    repository: Repository = dependencies["repository"]
-
-    request_params = ConsumerRequestParams(**event.queryStringParameters or {})
-
-    assert_no_extra_params(
-        request_params=request_params, provided_params=event.queryStringParameters
-    )
-
-    nhs_number: RequestQuerySubject = request_params.nhs_number
-
-    custodian = custodian_filter(
-        custodian_identifier=request_params.custodian_identifier
-    )
-
-    validate_type_system(request_params.type, pointer_types=data["pointer_types"])
-
-    pointer_types = type_filter(
-        type_identifier=request_params.type,
-        pointer_types=data["pointer_types"],
-    )
-
-    next_page_token: NextPageToken = request_params.next_page_token
-
-    if next_page_token is not None:
-        next_page_token = next_page_token.__root__
-
-    pk = key("P", nhs_number)
 
     count = 0
     next_page_token: NextPageToken = None
+
+    request_params = CountRequestParams(**event.queryStringParameters or {})
+
     while True:
-        response: PaginatedResponse = repository.query_gsi_1(
-            pk=pk,
-            type=pointer_types,
-            nhs_number=nhs_number,
-            exclusive_start_key=next_page_token,
-            limit=COUNT_ITEM_LIMIT,
+        response: PaginatedResponse = get_paginated_document_references(
+            data,
+            request_params,
+            event.queryStringParameters,
+            dependencies,
+            logger,
+            COUNT_ITEM_LIMIT,
+            next_page_token,
```

Comment:
for a function with this many params we should be using keyword arguments - can come back to bite us if we need to refactor positional args